for SG-31031 adding documentation for custom events, throttling (#436)
shotgunsoftware/developer.shotgunsoftware.com: docs/en/guides/webhooks.md

Your consumer response times to deliveries will impact webhooks throughput for your site.

Each site is allowed 1 minute of response time per minute. If all configured consumer endpoints for a site take the full 6 seconds to respond, webhooks deliveries for that site will be throttled to 10 per a minute.

When a high rate of overall throughput is needed, consumer endpoints should be designed according to the following model:
 1. Receive the request
 2. Spawn another process/thread to handle it the way you want
 3. Answer an acknowledging 200 immediately

Factors that may contribute to poor performance include:
- A poorly configured or under-resourced Webhook consumer endpoint
- Delays in processing the delivery before sending a response

#### Bursts

If your {% include product %} site is being throttled as a result of short bursts of heavy activity, it will return to normal throughput once the burst of event activity has subsided. The [performance indicators](#performance) will provide insight into which of your configured endpoints are not performing well.

#### Webhooks and geographic considerations

> The {% include product %} Webhook delivery infrastructure is hosted in the AWS US-East (Oregon) region. Optimization for delivery times may be possible by hosting your webhook endpint in a location closer to US-East.

Reviewer: ```suggestion
The {% include product %} Webhook delivery infrastructure is hosted in the AWS US-East (Oregon) region. Optimization for delivery times may be possible by hosting your Webhook endpint in that general location.
```

> The default {% include product %} Webhook delivery infrastructure is hosted in the AWS US-East (Oregon) region. Optimization for delivery times may be possible when [configuring your site to a location that is remote from this region](https://help.autodesk.com/view/SGSUB/ENU/?guid=SG_Administrator_ar_data_management_ar_storage_location_html). 

Author: @shaynacohen, I think my initial paragraph was confusing. it seems that the customer confusion was about where they located their web site/service to handle / respond to the deliveries. Since the EP service lives in US-East, if you put your webhook endpoint in Sydney, AUS. you would experience delays.  So we wouldn't want to link to the media S3 page.

> The default {% include product %} Webhook delivery infrastructure is hosted in the AWS US-East (Oregon) region. Optimization for delivery times may be possible when [configuring your site to a location that is remote from this region](https://help.autodesk.com/view/SGSUB/ENU/?guid=SG_Administrator_ar_data_management_ar_storage_location_html). 

Author: "A mention of geographic location of where events are coming from (US-EAST). We suffered from throttling immediately (our event response time was well exceeding the 60s of allotted time, up to 15 mins per 60s at times). While I do think that the 60s throttler is a bit tricky to work within, these were the crucial steps we had to take to ensure we didn't go past the limit.

 2. Spawn another process/thread to handle it the way you want
 3. Answer an acknowledging 200 immediately

> Factors that may contribute.

Author: fix this... contribute to what ?

Reviewer: How does it look with the update?
